src/online-stats.jl

```diff
@@ -0,0 +1,108 @@
+using OnlineStatsBase
+using StatsBase
+using OnlineStats
+
+import OnlineStats: Series
+
+import OnlineStatsBase: merge, AbstractSeries, OnlineStat
+
+export aggregate_stats
+
+"""
+`Series(xs::DArray, stats::OnlineStat...)`
+
+Create an `OnlineStats.Series` object with some initial data
+"""
+function Series(xs::DArray, stats::OnlineStat...)
+    function inner_series(xc)
+        Series(xc, map(copy, stats)...)
+    end
+
+    chunk_aggs = map(delayed(inner_series), xs.chunks)
+    collect(treereduce(delayed(merge), chunk_aggs))
+end
+
+function Series(xs::DArray, ys::DArray, stats::OnlineStat...)
+    function inner_series(xc, yc)
+        Series(xc, yc, map(copy, stats)...)
+    end
+
+    chunk_aggs = map(delayed(inner_series), xs.chunks, ys.chunks)
+    collect(treereduce(delayed(merge), chunk_aggs))
+end
+
+function Series(t::Union{DTable,IndexedTable}, stats::OnlineStat...)
+    Series(values(t), stats...)
+end
+
+function Series(t::IndexedTable, stats::OnlineStat...)
+    Series(values(t), stats...)
+end
+
+# spcialization for linear regression
+function Series(x::Columns, y::AbstractVector, stats::OnlineStat{(1,0), 1}...)
+    xmatrix = getindex.([columns(x)...]', 1:length(x))
+
+    Series(xmatrix, y, stats...)
+end
+
+"""
+`aggregate_stats(series::AbstractSeries, t::Union{IndexedTable, DTable)`
+
+Aggregate common indices with an `OnlineStas.Series` object.
+
+Computes the given Online stat for every group of values with equal indices.
+"""
+function aggregate_stats(series::AbstractSeries, t::Union{IndexedTable, DTable})
+    aggregate_stats(series, keys(t), values(t))
+end
+
+@inline function _fit!(series, xs::Tup, y...)
```

Comment:
Pretty sure this only helps give correct input to `LinReg`. It would be nice to have a `fit!` method which takes a tuple and a float for LinReg rather than a vector and float. Can avoid a ton of allocation. @joshday

Reply:
I must have missed LinReg.  Everything else that uses a vector observation should accept a tuple already.  Fixed on master.

Reply:
Thank you!